api/server.go

```diff
@@ -48,28 +50,58 @@ func NewServer(config common.Config) error {
 		return errors.New("'org' cannot be empty in the configuration")
 	}
 	Org = config.Org
-	repo := config.Repository
+	metadata := config.MetadataRepository
 
-	// Create metadata repository session
-	log.Debugf("Creating new MetadataRepository of type %s with configuration %+v (org: %s)", repo.Type, repo.Config, Org)
+	// Initialize metadata repository session
+	log.Debugf("Creating new session for MetadataRepository of type %s with configuration %+v (org: %s)", metadata.Type, metadata.Config, Org)
 
 	var metadataRepo dataset.MetadataRepository
 	var err error
-	switch repo.Type {
+	switch metadata.Type {
 	case "s3":
-		metadataRepo, err = s3metadatarepository.NewDefaultRepository(repo.Config)
+		if metadata.Config["prefix"] == nil {
+			metadata.Config["prefix"] = Org
+		} else {
+			metadata.Config["prefix"] = metadata.Config["prefix"].(string) + "/" + Org
+		}
+		metadataRepo, err = s3metadatarepository.NewDefaultRepository(metadata.Config)
 		if err != nil {
 			return err
 		}
 	default:
-		return errors.New("failed to determine metadata repository type, or type not supported: " + repo.Type)
+		return errors.New("failed to determine metadata repository type, or type not supported: " + metadata.Type)
 	}
 
-	// Create a shared session
-	for name, c := range config.Accounts {
-		log.Debugf("Creating new service for account '%s' with key '%s' in region '%s' (org: %s)", name, c.Akid, c.Region, Org)
+	// Create dataset service sessions
+	for name, a := range config.Accounts {
+		creds := map[string]interface{}{
+			"region":   a.Config.Region,
+			"akid":     a.Config.Akid,
+			"secret":   a.Config.Secret,
+			"endpoint": a.Config.Endpoint,
+		}
```

Comment:
Why not make the Account.Config a `map[string]interface{}` instead of doing this?  after all, its more than credentials.